sw/device/lib/dif/dif_spi_host.c

```diff
@@ -0,0 +1,393 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_host.h"
+
+#include <assert.h>
+#include <stdalign.h>
+#include <stddef.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+
+#include "spi_host_regs.h"  // Generated.
+
+// We create weak symbol aliases for the FIFO write and read functions so the
+// unit tests can provide mocks.  The mocks provide for separate testing of
+// the FIFO functions and the overall transaction management functions.
+OT_WEAK
+OT_ALIAS("dif_spi_host_fifo_write")
+void spi_host_fifo_write_alias(const dif_spi_host_t *spi_host, const void *src,
+                               uint16_t len);
+
+OT_WEAK
+OT_ALIAS("dif_spi_host_fifo_read")
+void spi_host_fifo_read_alias(const dif_spi_host_t *spi_host, void *dst,
+                              uint16_t len);
+
+static void spi_host_reset(const dif_spi_host_t *spi_host) {
+  // Set the software reset request bit.
+  mmio_region_write32(
+      spi_host->base_addr, SPI_HOST_CONTROL_REG_OFFSET,
+      bitfield_bit32_write(0, SPI_HOST_CONTROL_SW_RST_BIT, true));
+
+  // Wait for the spi host to go inactive.
+  bool active;
+  do {
+    uint32_t reg =
+        mmio_region_read32(spi_host->base_addr, SPI_HOST_STATUS_REG_OFFSET);
+    active = bitfield_bit32_read(reg, SPI_HOST_STATUS_ACTIVE_BIT);
```

Comment:
Personally I'd have written this as `if(!read()) { break }`. But it's fin as it is.